src/components/ContributePrompt.js

```diff
@@ -79,7 +79,7 @@ export default class ContributePrompt extends Component {
                     {session.isAnonymous ? 'anonymous' : `@${session.username}`}
                   </FormControl.Static>{' '}
                   {session.isAnonymous && (
-                    <Button bsStyle="success" onClick={onLogin}>
+                    <Button bsStyle="success" className="contributePrompt__modal--loginButton" onClick={onLogin}>
```

Comment:
Should likely note the BEM naming model use in the coding style doc somewhere.

Should we also tweak these names to indicate that they are not "real" styles? AFAICT we would never want to actually have a style called that. Rather we are adding these for identification/testing purposes. My expectation is that our styling is more macro with rare exceptions where we tweak just one element. For example, we'd figure out the styling for a general class of UI scenarios and then apply that everywhere (e.g., "spaced-out-list") rather than driving the style of each instance individually.

Make sense?

Reply:
I got your point, it depends if we want to use this identifiers only for testing purposes or to attach a real class that can be customized.
The BEM notation helps to identify some specific elements (ex. we can have multiple loginButton in the same view, but i want to test only the one in the contribute modal).

We can address the same thing using different approaches, like using ids or attributes (like name).
We should keep a strategy and go on.

Thoughts?

Reply:
Re: discord conversation:

This class name itself is a little wild, but it's obvious enough visually that I'd not personally try to style something against. Class names are a good identifier for that purpose, and don't _necessarily_ have semantic value. It does put some extra data into the DOM, but it's very minimal.

If you're looking for an XPath approach, selectors like this could be useful:

```
//div[@class="modal"]//button[text()="Login"]
```

This would select a button with the text of "Login" within a div (anywhere in the DOM) with a class of "modal". `//` is essentially a relative search; you can think of it as `/**/` with glob-style patterns.

Note that `class` doesn't have any special meaning in an XPath query; it's one string and not a set of classes. A more "correct" expression would be:

```
//div[contains(concat(' ',normalize-space(@class),' '), 'modal']//button[text()="Login"]
```

That allows "modal" as one of the classes, not the only class. I didn't verify this exact expression against the code in this PR; but hopefully you get the idea.

Great XPath resource here: https://devhints.io/xpath and your browser console can likely evaluate these as well. In Firefox `$x` is a function to evaluate/play around with expressions in the console. So yeah, it's kinda gnarly. XPath expressions are super useful..... but you could also get away just fine with CSS classnames like this one. :)

Reply:
So from the above and some of the discussion in Discord, using the classes seems easy and attainable. The BEM style names are pretty verbose but its a known convention so why invent another unless we need to. The only tweak I'd put on is *perhaps* a `t_` prefix or something to indicate that this is not actually a class, just a name. 

While I agree with @jpeddicord that the names here are unattractive enough that folks would likely not try to style, I worry about test oriented names higher up in the dom. For example, `contributePrompt`. (well, `contribute-prompt`) That is attractive as an actual style class. `t_contribute-prompt` (or something) is different enough to give pause.

Having said all of that, it feels "yucky" to be sprinkling these long class names all over the code. https://ghostinspector.com/blog/css-selector-strategies-automated-browser-testing/ has an interesting hierarchy in the "How do I know when to use the different methods?"How do I know when to use the different methods? section that perhaps we can use to guide us here. The main point here is to use something reasonably stable. Something that is already there would be great.

Reply:
I've read the article and it sounds fair enough. I will try to change some identifiers logic in order to have less specific class names that could create some confusion to the developers.

Reply:
I've applied most of the rules, but, anyway, a point to consider is that if a user wants to edit the style of a specific element, that could be done even without any specific classname, using attributes or ids, or style prop directly.
That is, using any existing element or class for tests purposes avoids to add other code to maintain, and that's perfectly fine, but doesn't discourage a user to customize the style, if he really wants.

Reply:
Oh absolutely. My point was not about "stopping" people doing it, rather not encouraging them to. If you walk up to a bit o DOM and it has some css classes identified, it's pretty reasonable to think "hey, I could scoot that element over a bit with a margin, or change the color by tweaking that class".  

Additionally, we are putting a class name there but don't actually have a class. It's good to point out that that is intentional. Otherwise I could see someone cleaning up and removing "unsued" classes.

It's just about making the code more self-evident.